tests/unit/checkboxradio/checkboxradio_core.js

```diff
@@ -0,0 +1,196 @@
+
+	// click in second form
+	$("#radio2 .ui-button:eq(0)").simulate("click");
+	assert(":eq(1)", ":eq(0)", ":eq(0)");
+});
+
+asyncTest(
+	"Checkbox/Radiobutton do not Show Focused State when using Keyboard Navigation",
+	function( assert ) {
+		expect( 2 );
+		var check = $( "#check" ).checkboxradio(),
+			label = $( "label[for='check']" );
+		assert.lacksClasses( label, "ui-state-focus" );
+		check.focus();
+		setTimeout(function() {
+			assert.hasClasses( label, "ui-state-focus" );
+			start();
+		});
+	}
+);
+
+asyncTest( "Ensure checked after single click on checkbox label button", function( assert ) {
+	expect( 2 );
+
+	$( "#check2" ).checkboxradio().change( function() {
+		var label = $( this ).checkboxradio( "widget" );
+		ok( this.checked, "checked ok" );
+
+		// The following test is commented out for now because with new markup we are trying to avoid aria
+		//ok( lbl.attr("aria-pressed") === "true", "aria ok" );
+		assert.hasClasses( label, "ui-state-active" );
+	});
+
+	// Support: Opera
+	// Opera doesn't trigger a change event when this is done synchronously.
+	// This seems to be a side effect of another test, but until that can be
+	// tracked down, this delay will have to do.
+	setTimeout(function() {
+		$( "#check2" ).checkboxradio( "widget" ).simulate( "click" );
+		start();
+	});
+});
+test( "Checkbox creation that requires a matching finds label in all cases", function( assert ) {
+	expect( 7 );
+	var group = $( "<span><label for='t7092a'></label><input type='checkbox' id='t7092a'></span>" );
+	group.find( "input[type=checkbox]" ).checkboxradio();
+	assert.hasClasses( group.find( "label" ), "ui-button" );
+
+	group = $( "<input type='checkbox' id='t7092b'><label for='t7092b'></label>" );
+	group.filter( "input[type=checkbox]" ).checkboxradio();
+	assert.hasClasses( group.filter( "label" ), "ui-button" );
+
+	group = $( "<span><input type='checkbox' id='t7092c'></span><label for='t7092c'></label>" );
+	group.find( "input[type=checkbox]" ).checkboxradio();
+	assert.hasClasses( group.filter( "label" ), "ui-button" );
+
+	group = $( "<span><input type='checkbox' id='t7092d'></span><span><label for='t7092d'></label></span>" );
+	group.find( "input[type=checkbox]" ).checkboxradio();
+	assert.hasClasses( group.find( "label" ), "ui-button" );
+
+	group = $( "<input type='checkbox' id='t7092e'><span><label for='t7092e'></label></span>" );
+	group.filter( "input[type=checkbox]" ).checkboxradio();
+	assert.hasClasses( group.find( "label" ), "ui-button" );
+
+	group = $( "<span><label><input type='checkbox' id='t7092f'></label></span>" );
+	group.find( "input[type=checkbox]" ).checkboxradio();
+	assert.hasClasses( group.find( "label" ), "ui-button" );
+
+	group = $( "<span><input type='checkbox' id='check:7534'><label for='check:7534'>Label</label></span>" );
+	group.find( "input" ).checkboxradio();
+	assert.hasClasses( group.find( "label" ), "ui-button", "Found an label with id containing a :" );
```

Comment:
These are very similar, only the markup changes and this last assertion has a message. Let's abstract these a bit, where we have an array of different markups and just iterate over that.